acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -520,25 +519,34 @@ def check_puppet_non_blocking_response(identity, transaction_id, max_retries, qu
         # allow a few unknowns before giving up
         unknown_count += 1
       else
-        puppet_run_result = JSON.parse(action_result['stdout'])['status']
-        puppet_run_environment = JSON.parse(action_result['stdout'])['environment']
+        run_result = JSON.parse(action_result['stdout'])
```

Comment:
Only if there was an error. I can add a test for it. I'm not really doing error testing on responses here, but unit tests cover them fairly well.